src/Core/OrganizationFeatures/OrganizationCollections/DeleteCollectionCommand.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Bit.Core.Entities;
+using Bit.Core.OrganizationFeatures.OrganizationCollections.Interfaces;
+using Bit.Core.Repositories;
+using Bit.Core.Services;
+
+namespace Bit.Core.OrganizationFeatures.OrganizationCollections;
+
+public class DeleteCollectionCommand : IDeleteCollectionCommand
+{
+    private readonly ICollectionRepository _collectionRepository;
+    private readonly IEventService _eventService;
+    public DeleteCollectionCommand(
+        ICollectionRepository collectionRepository,
+        IEventService eventService)
+    {
+        _collectionRepository = collectionRepository;
+        _eventService = eventService;
+    }
+    public async Task DeleteAsync(Collection collection)
+    {
+        await _collectionRepository.DeleteAsync(collection);
+        await _eventService.LogCollectionEventAsync(collection, Enums.EventType.Collection_Deleted);
```

Comment:
question: on row 41 you are calling the method with a `deleteDate`, but on this row you are not, did you miss it? Or maybe I'm the one missing something 😄